torchci/lib/benchmark/llms/utils/llmUtils.ts

```diff
@@ -145,206 +225,263 @@ export function combineLeftAndRight(
     }
 
-    if ("metadata" in row) {
-      data.push(row);
+    if (
+      repoName === "pytorch/pytorch" &&
+      benchmarkName === "TorchCache Benchmark"
+    ) {
+      const extraInfo = JSON.parse(extra);
+      row["is_dynamic"] = extraInfo["is_dynamic"];
     }
-  });
 
-  return data;
-}
+    if (metric == "FAILURE_REPORT") {
+      row[metric] = {
+        l: hasL
+          ? {
+              actual: Number.MAX_SAFE_INTEGER, // indicate the failure on left side
+              target: 0,
+            }
+          : {
+              actual: 0,
+              target: 0,
+            },
+        r: hasR
+          ? {
+              actual: Number.MAX_SAFE_INTEGER, // indicate the failure on right side
+              target: 0,
+            }
+          : {
+              actual: 0,
+              target: 0,
+            },
+        highlight: hasL && hasR,
+      };
+    } else {
+      row[metric] = {
+        l: hasL
+          ? {
+              actual: record["l"].actual,
+              target: record["l"].target,
+            }
+          : {
+              actual: 0,
+              target: 0,
+            },
+        r: hasR
+          ? {
+              actual: record["r"].actual,
+              target: record["r"].target,
+            }
+          : {
+              actual: 0,
+              target: 0,
+            },
+        highlight: hasL && hasR,
+      };
+    }
+  }
+  return row;
+};
 
-export function computeGeomean(data: LLMsBenchmarkData[], metricName: string) {
-  const metricValues: { [key: string]: number[] } = {};
-  const returnedGeomean: LLMsBenchmarkData[] = [];
+const processJobLevelFailureRows = (
+  dataGroupedByModel: { [k: string]: any },
+  repoName: string
+): Set<string> => {
 
-  data.forEach((r: LLMsBenchmarkData) => {
-    if (r.metric !== metricName) {
-      return;
-    }
 
-    const origins = r.origins.join(",");
-    const k = `${r.granularity_bucket}+${r.workflow_id}+${r.job_id}+${r.backend}+${r.dtype}+${origins}+${r.device}+${r.arch}+${r.metric}`;
-    if (!(k in metricValues)) {
-      metricValues[k] = [];
+  // see if a repo need special handling for job level failure
+  const config = getJobReportFailureConfigs();
+  if (!(repoName in config)) {
+    return new Set();
+  }
+
+  const repoSpecificConfig: any = config[repoName];
+  const jobLevelFailureConfig = repoSpecificConfig["job_level_failure"];
+
+  // find rows that related to the job level failure
+  const jobLevelFailureKeys = Object.keys(dataGroupedByModel).filter(
+    (key: string) => {
+      const identifier = jobLevelFailureConfig["key_name"];
+      const val = getGroupKeyItem(key, identifier);
+      const extra = getGroupKeyItem(key, "extra");
+      const extraInfo = JSON.parse(extra);
+
+      const isJobLevelFailure = extraInfo["failure_type"] === "GIT_JOB";
+      if (!val) {
+        return false;
+      }
+      if (jobLevelFailureConfig["content"].includes(val) && isJobLevelFailure) {
```

Comment:
Umm, do you know that is the default value returning by `filter` here? I'm trying to skim through https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter but haven't found it yet.  I'm trying to figure out the behavior when a new device is added into the list:

```
apple_iphone_15,
samsung_galaxy_s22,
samsung_galaxy_s24,
google_pixel_8_pro,
```

where `jobLevelFailureConfig["content"].includes(val)` would return false.